tests/components/evohome/conftest.py

```diff
@@ -153,3 +158,20 @@ async def setup_evohome(
         assert mock_client.account_info is not None
 
         yield mock_client
+
+
+def get_dhw_entity(hass: HomeAssistant) -> EvoDHW | None:
+    """Return the DHW entity of the evohome system."""
+
+    broker: EvoBroker = hass.data[DOMAIN]["broker"]
+
+    if (dhw := broker.tcs.hotwater) is None:
+        return None
+
+    entity_registry = er.async_get(hass)
+    entity_id = entity_registry.async_get_entity_id(
+        Platform.WATER_HEATER, DOMAIN, dhw._id
+    )
+
+    component: EntityComponent = hass.data.get(Platform.WATER_HEATER)  # type: ignore[assignment]
+    return next(e for e in component.entities if e.entity_id == entity_id)  # type: ignore[return-value]
```

Comment:
This is specific code to `water_heater` should it not be inclusde in `test_water_heater.py` ?
Another option is to make this a fixture parameterize the platform with as a second fixture, so it can be reused.